src/spi.rs

```diff
@@ -1051,21 +1051,32 @@ where
     }
 
     fn send(&mut self, byte: u8) -> nb::Result<(), Error> {
-        let sr = self.spi.sr.read();
+        // Read from DR register to clear OVR
+        let _ = self.spi.dr.read();
```

Comment:
I would feel more comfortable if we only did this read upon overrun detection. I'm curious if there's a potential race condition here of reading the DR right as we finish shifting out the current TX byte (e.g. we have received a byte for the current transfer, but haven't quite finished sending ours?). Not sure if this can really happen, but it wouldn't hurt to be careful.